tests/solana_tests/primitives.rs

```diff
@@ -599,6 +604,363 @@ fn truncate_uint(n: &mut U256, width: usize) {
+                    width.try_into().unwrap(),
+                )),
+                ethabi::Token::Int(bigint_to_eth(&second_op, width.try_into().unwrap())),
+            ],
+            &[],
+            None,
+        );
+
+        let res = first_operand_rand.mul(second_op);
+        assert_eq!(
+            return_value,
+            vec![ethabi::Token::Int(bigint_to_eth(
+                &res,
+                width.try_into().unwrap(),
+            ))]
+        );
+    }
+}
+
+#[test]
+fn test_mul_within_range() {
+    let mut rng = rand::thread_rng();
+    for width in (8..=256).step_by(8) {
+        let src = r#"
+        contract test {
+            function mul(uintN a, uintN b) public returns (uintN) { 
+                return a * b;
+            }
+        }"#
+        .replace("intN", &format!("int{}", width));
+
+        let mut contract = build_solidity(&src);
+        contract.constructor("test", &[]);
+        for _ in 0..10 {
+            // Max number to fit unsigned N bits is 0..(2^N)-1
+            let limit = BigUint::from(2_u32).pow(width).sub(1_u32);
+
+            // Generate a random number within the the range 0..2^N -1
+            let first_operand_rand = rng.gen_biguint(width.into());
+
+            // Calculate a number that when multiplied by first_operand_rand, the result will not overflow N bits (the result of this division will cast the float result to int result, therefore lowering it. The result of multiplication will never overflow).
+            let second_operand_rand = limit.div(&first_operand_rand);
+
+            let return_value = contract.function(
+                "mul",
+                &[
+                    ethabi::Token::Uint(U256::from_big_endian(&first_operand_rand.to_bytes_be())),
+                    ethabi::Token::Uint(U256::from_big_endian(&second_operand_rand.to_bytes_be())),
+                ],
+                &[],
+                None,
+            );
+            let res = first_operand_rand * second_operand_rand;
+
+            assert_eq!(
+                return_value,
+                vec![ethabi::Token::Uint(U256::from_big_endian(
+                    &res.to_bytes_be()
+                ))]
+            );
+        }
+    }
+}
+
+#[test]
+fn test_overflow_detect_signed() {
+    let mut rng = rand::thread_rng();
+    // For bit sizes from 8..64, LLVM has intrinsic functions for multiplication with overflow. Testing starts from int types of 72 and up. There is no need to test intrinsic LLVM functions.
+    for width in (72..=256).step_by(8) {
+        let src = r#"
+        contract test {
+            function mul(intN a, intN b) public returns (intN) {
+                return a * b;
+            }
+        }"#
+        .replace("intN", &format!("int{}", width));
+        let mut contract = build_solidity(&src);
+
+        contract.constructor("test", &[]);
+
+        // The max value held in signed N bits is - 2^(N-1)..2^(N-1)-1 .Generate a value that will overflow this range:
```

Comment:
```suggestion
        // The max value held in signed N bits is [-2^(N-1), 2^(N-1)-1]. Generate a value that will overflow this range:
```
The mathematical notation for range is clearer.